pkg/k8sclient/types.go

```diff
@@ -30,7 +30,7 @@ import (
 const bytesToKb = 1024
 
 // Used to guard access to the pod, task and job related maps.
-var PodsCond *sync.Cond
+var PodsMux *sync.RWMutex
```

Comment:
@shivramsrivastava 
Shouldn't `PodMux` be a part of `PodWatcher` struct instead? I don't see this being used anywhere else. 

Plus what about other maps below? can they also be moved inside `PodWatcher` ?

```
 var PodToTD map[PodIdentifier]*firmament.TaskDescriptor
 var TaskIDToPod map[uint64]PodIdentifier
 var jobIDToJD map[string]*firmament.JobDescriptor
```

Reply:
@dhilipkumars 
The maps and the mutexes are accessed across packages like stats package, k8sclient, and the main package. So they are global.

The stats package use these global mutexes to lock and read/update the pods and nodes from the already populated maps. If the map doesn't have an entry for the pod or node the stats package will not update the pod or node info.

 **About stats module:**
    Very briefly stats module gets the info from heaspter and filters what is required and send that information to the firmament's knowledgebase.

So we had kept them as global.

> I will think of something to make them closer to where they belong, for clarity.